pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/fst/FstIndexType.java

```diff
@@ -167,4 +173,10 @@ protected TextIndexReader createIndexReader(PinotDataBuffer dataBuffer, ColumnMe
       return createIndexReader(dataBuffer, metadata);
     }
   }
+
+  @Override
+  public void convertToNewFormat(TableConfig tableConfig, Schema schema) {
+    super.convertToNewFormat(tableConfig, schema);
```

Comment:
Here is a nice way to avoid having to do the super call in every implementation, I think it applies here and makes code more maintainable:[here](https://stackoverflow.com/questions/10635279/call-super-class-method-automatically)

Reply:
That is more or less the pattern we follow with `getConfig` and `createDeserializer`. I slightly prefer what @jugomezv suggest, but IMHO it is not a blocker.

Reply:
I liked what @jugomezv suggested as mostly the additional override logic is around cleaning the old properties in table config which are specific to an index type. Will make the changes.